test/powershell/engine/ETS/Adapter.Tests.ps1

```diff
@@ -1,143 +1,228 @@
-                {
-                    return 1;
+    Context "Property Adapter Tests" {
+        BeforeAll {
+            $pso  = [System.Diagnostics.Process]::GetCurrentProcess()
+            $processName = $pso.Name
+
+            if(-not ('TestCodeMethodClass' -as "type"))
+            {
+                class TestCodeMethodClass {
+                    static [int] TestCodeMethod([PSObject] $target, [int] $i)
+                    {
+                        return 1;
+                    }
                 }
             }
+
+            $psmemberset = new-object System.Management.Automation.PSMemberSet 'setname1'
+            $psmemberset | Add-Member -MemberType NoteProperty -Name NoteName -Value 1
+            $testmethod = [TestCodeMethodClass].GetMethod("TestCodeMethod")
+            $psmemberset | Add-Member -MemberType CodeMethod -Name TestCodeMethod -Value $testmethod
+
+            $document = new-object System.Xml.XmlDocument
+            $document.LoadXml("<book ISBN='12345'><title>Pride And Prejudice</title><price>19.95</price></book>")
+            $doc = $document.DocumentElement
+        }
+        It "can get a Dotnet parameterized property" {
+            $col  = $pso.psobject.Properties.Match("*")
+            $prop = $col.psobject.Members["Item"]
+            $prop | Should Not BeNullOrEmpty
+            $prop.IsGettable | Should be $true
+            $prop.IsSettable | Should be $false
+            $prop.TypeNameOfValue | Should be "System.Management.Automation.PSPropertyInfo"
+            $prop.Invoke("ProcessName").Value | Should be $processName
         }
 
-        $psmemberset = new-object System.Management.Automation.PSMemberSet 'setname1'
-        $psmemberset | Add-Member -MemberType NoteProperty -Name NoteName -Value 1
-        $testmethod = [TestCodeMethodClass].GetMethod("TestCodeMethod")
-        $psmemberset | Add-Member -MemberType CodeMethod -Name TestCodeMethod -Value $testmethod
+        It "can get a property" {
+            $pso.psobject.Properties["ProcessName"] | should Not BeNullOrEmpty
+        }
 
-        $document = new-object System.Xml.XmlDocument
-        $document.LoadXml("<book ISBN='12345'><title>Pride And Prejudice</title><price>19.95</price></book>")
-        $doc = $document.DocumentElement
-    }
-    It "can get a Dotnet parameterized property" {
-        $col  = $pso.psobject.Properties.Match("*")
-        $prop = $col.psobject.Members["Item"]
-        $prop | Should Not BeNullOrEmpty
-        $prop.IsGettable | Should be $true
-        $prop.IsSettable | Should be $false
-        $prop.TypeNameOfValue | Should be "System.Management.Automation.PSPropertyInfo"
-        $prop.Invoke("ProcessName").Value | Should be $processName
-    }
+        It "Can access all properties" {
+            $props = $pso.psobject.Properties.Match("*")
+            $props | should Not BeNullOrEmpty
+            $props["ProcessName"].Value |Should be $processName
```

Comment:
Since the test is accessing all properties, it seems like we should validate the property count.

Reply:
I don't know that is expected count.

Reply:
It's probably safe to maybe check GreaterThan 1?